addon/services/metrics.js

```diff
@@ -49,8 +57,13 @@ export default class Metrics extends Service {
    * @return {Void}
    */
   init() {
-    const adapters = this.options.metricsAdapters || emberArray();
     const owner = getOwner(this);
+    const config = owner.factoryFor('config:environment').class;
```

Comment:
does this still work properly with embroider?

Reply:
I do not know. Not sure how to test.

The original import of ../config/environment did not see to work correctly when running in tests if I tried to use that in the service

Reply:
Yes this does work in embroider, safe, the addon doesnt work under optimized for various reasons, not sure if this is one of them or not. I created another PR to add embroider try scenarios https://github.com/adopted-ember-addons/ember-metrics/pull/280 although I dont understand why some of the checks were cancelled.

Reply:
@ef4 any thoughts on this?

Reply:
This part is fine. 

The part of this addon that isn't going to work with optimizations is `_lookupAdapter`. IMO, the simplest way to make it work would be to change the API so that users configure their adapters from somewhere in their own Javascript, and they import the adapters they want instead of using string names for them. Something like:

```js
// app.js
import { configureMetricsAdapter } from 'ember-metrics';
import GoogleAnalyticsAdapter from 'ember-metrics/metrics-adapters/google-analytics';

configureMetricsAdapter(GoogleAnalyticsAdapter, { optionsGoHere });
```

Alternatively, you could probably keep the adapter config as it is, but send it into `@embroider/macros` `setOwnConfig` and use it to emit code in the addon that imports only the adapters that are needed.

Reply:
The point of this PR was to fix the deprecation of no implicit injection. Not to make it embroider compatible. I think the question was if the one change I made was compatible and Ed confirmed it is.

For a PR to make it embroider compatible Ed is absolutely correct that the lookup adapter would fail unless the consuming app did something. I would suggest that the app create a a metrics adapter in the following file `metrics-adapters/google-analytics.js` containing 
```
import BaseAdapter from 'ember-metrics/metrics-adapters/google-analytics';

export default class GoogleAnalytics extends BaseAdapter {
}

```

The app only creates the ones it uses, it gives a clear place where they can override options if needed and requires no code change to this addon. Just an update to the docs that you dont have to create empty exports like this in ember classic or embroider safe, but you do if you want your app to be embroider optimized